### DISCUSSION TITLE
Create button not shown after upgrade from filament 2 to 3

### DISCUSSION THREAD

**Discussion by jay2success** [filamentphp/filament#8715] [Help] [answered]
Hello,

I am doing upgrade from filament 2 to 3.
After finished, the admin panel shows fine but the create button is not visible.
The model policy already return true so no issue with authorization.
It's showing fine when i change the url directly: "https://localhost/admin/packages/create"
But create button not shown on list page: https://localhost/admin/packages
All resources has same issue.
Please help. 
Thanks.

**Comment by ousid** [comment]
Did you point the model with the policy in `AuthServiceProvider`?

Try to clear the cache also `php artisan cache:clear`

Try to delete one of the `ModelPolicy` classes, and see if the create button returns or not. If yes, you have an issue, in regard to the `ModelPolicy`, if not, you should be else where.

**Comment by jay2success** [comment]
Hi,

Thanks for your answer.

The policy already using auto discovery in AuthServiceProvider so there is no issue with it.

Already tried to delete the model Policy and issue still there.

I even tried to skip the policy in filament resource using this code and its still not working
`protected static bool $shouldSkipAuthorization = true;`

I already cleared cache many times and still has issue.

I tried to check the views on "vendor\filament\filament\resources\views\components\header\index.blade.php"
The "actions" prop contains empty value and i have no idea where the actions set from, it should be automatic like on filament 2.

So any more idea how to solve this?

**Comment by jay2success** [comment, marked as answer]
Hi,

Ok i solved it.
on List page, i have to add this code:

```
 protected function getHeaderActions(): array
    {
        return [
            Actions\CreateAction::make(),
        ];
    }
```

On filament 2, its automatically shown without this code, not sure why filament 3 has to add this code.
Thanks.
